### PR TITLE
feat(cfg): env secret/string support

### PR DESCRIPTION
**Pull request by AndrewWinterman**
Please read [CONTRIBUTING.md](CONTRIBUTING.md) for additional information on contributing to this repository!

## Jira ID

[XX-XX]

### CODE REVIEW

File: pkg/cfg/env_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+)
+
+func TestEnvString(t *testing.T) {
+	key := uuid.NewString()
+	value := uuid.NewString()
+	notsetKey := uuid.NewString()
+	err := os.Setenv(key, value)
+	if err != nil {
+		t.Fatalf("failed to set environment variable: %v", err)
+	}
```
Comment (malept) [inline pkg/cfg/env_test.go:17]: suggestion: I would use `gotest.tools/v3/assert` here.
```suggestion
	assert.NilError(t, err, "failed to set environment variable")
```

File: pkg/cfg/env_test.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+	t.Run("returns set value", func(t *testing.T) {
+		v, err := EnvString(key)
+		if err != nil {
+			t.Fatalf("got unexpected error %s", err.Error())
+		}
+
+		if v != value {
+			t.Fatalf("expected %s; got %s", value, v)
+		}
```
Comment (malept) [inline pkg/cfg/env_test.go:34]:
```suggestion
		assert.Equal(t, v, value)
```

File: pkg/cfg/env_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+func TestEnvSecret(t *testing.T) {
+	key := uuid.NewString()
+	value := uuid.NewString()
+	notsetKey := uuid.NewString()
+	err := os.Setenv(key, value)
+	if err != nil {
+		t.Fatalf("failed to set environment variable: %v", err)
+	}
```
Comment (malept) [inline pkg/cfg/env_test.go:45]:
```suggestion
	assert.NilError(t, err, "failed to set environment variable")
```

File: pkg/cfg/env_test.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+	t.Run("if key not set; error is returned", func(t *testing.T) {
+		v, err := EnvSecret(notsetKey)
+		if err == nil {
+			t.Fatalf("expected error fetching %s; got nil error and value %v", notsetKey, v)
+		}
```
Comment (malept) [inline pkg/cfg/env_test.go:51]:
```suggestion
		t.ErrorContains(t, err, notsetKey)
		t.ErrorContains(t, err, "environment variable not set")
```

File: pkg/cfg/env_test.go [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	})
+
+	v, err := EnvSecret(key)
+	if err != nil {
+		t.Fatalf("got unexpected error %s", err.Error())
+	}
```
Comment (malept) [inline pkg/cfg/env_test.go:57]:
```suggestion
	assert.NilError(t, err)
```

File: pkg/cfg/env_test.go [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+	if string(v) != value {
+		t.Fatalf("expected %s; got %s", value, v)
+	}
```
Comment (malept) [inline pkg/cfg/env_test.go:61]:
```suggestion
	assert.Equal(t, string(v), value)
```

**Review by malept — CHANGES_REQUESTED** [review]
This seems mostly fine but the tests should be using `gotest.tools/v3/assert`.

File: pkg/cfg/env.go [hunk 7]
```diff
@@ -0,0 +1,33 @@
+// Copyright 2022 Outreach Corporation. All Rights Reserved.
```
Comment (malept) [inline pkg/cfg/env.go:1]: Nit:
```suggestion
// Copyright 2024 Outreach Corporation. All Rights Reserved.
```

**Review by malept — APPROVED** [review]
Looks fine once the copyright date is changed.

**Review by daniel-dostal84 — APPROVED** [review]

**Pull request merged**